codice/ddf
# DDF-5080 Provide an extension point for search-interactions in Intrigue

#### What does this PR do?
Provide an extension point for search-interactions in Intrigue. Allow downstream projects to add custom search-interactions.
Also adds a function to `lightbox` that can be optionally set and will be called when `lightbox` is closed
#### Who is reviewing it? 
@willwill96 @hayleynorton @zta6 @Corey-Collins @bellcc 
#### Select relevant component teams: 
@codice/ui 
#### Ask 2 committers to review/merge the PR and tag them here.

@adimka 
@djblue

#### How should this be tested?
- build / run
- In a new search form, verify no regression in the interaction dropdown menu (see screenshot). specifically switching between forms (including user defined forms) and reseting the form. 
#### Any background context you want to provide?
#### What are the relevant tickets?
Fixes: #5080

#### Screenshots
<img width="518" alt="Screen Shot 2019-07-10 at 1 36 31 PM" src="https://user-images.githubusercontent.com/39923178/60999180-c23f1480-a317-11e9-8d71-3574bd90d0a9.png">

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
